src/gt4py/stencil_builder.py

```diff
@@ -18,6 +18,7 @@
 from typing import TYPE_CHECKING, Any, Dict, Optional, Type, Union
 
 import gt4py.caching
+import gt4py.frontend
```

Comment:
Good catch! I know it's outside of the scope of this PR, but if everything was working before, is this really needed? And if it is needed, why is not tested?

Reply:
I have to admit I do not understand why this did not surface elsewhere/before. The circular import surfaced in `conftest.py` which should be run everytime tests are executed. Maybe another import avoiding this happens before occasionally? Anyway running `python ./tests/conftest.py` also fails on the current master.

Reply:
I don't think it is ever expected to run `conftest.py` as a script, it's mostly just a container of pytest settings.

Reply:
I see. For me the error occured when executing the tests with PyCharm